### PR TITLE
feat: :sparkles: Allow overwriting fs layout; rename SERVER_DATA_MANAGER to SESSION_MANAGER

### PR DESCRIPTION
**Pull request by zarik5**
This PR is required for Monado support, as in that case the FILESYSTEM_LAYOUT should not be initialized using the openvr config.
Furthermore this PR allows ServerCore to be initialized without requiring fs I/O, so it can be easily integrated in unit tests.

### CODE REVIEW

File: alvr/server_core/src/web_server.rs [hunk 1]
```diff
@@ -267,37 +270,7 @@ async fn http_api(
                 .body(latency.to_string().into())?
         }
         "/api/ping" => reply(StatusCode::OK)?,
-        other_uri => {
```
Comment (Meister1593) [inline alvr/server_core/src/web_server.rs:270]: why did this exist before? is it for wasm?

Comment (Meister1593) [reply alvr/server_core/src/web_server.rs:270]: if it's unused now, i would delete it tbh

Comment (zarik5) [reply alvr/server_core/src/web_server.rs:270]: Ok then

File: alvr/server_core/src/connection.rs [hunk 2]
```diff
@@ -397,13 +397,13 @@ fn connection_pipeline(
 ) -> ConResult {
     // This session lock will make sure settings cannot be changed while connecting and no other
     // client can connect (until handshake is finished)
-    let mut server_data_lock = SERVER_DATA_MANAGER.write();
+    let mut session_manager_lock = SESSION_MANAGER.write();
```
[the remark this answers is not in the document]
Comment (zarik5) [reply alvr/server_core/src/connection.rs:401]: I just expanded a little bit, but not sure how to explain it better, seems to be already clear.

**Review by The-personified-devil — APPROVED** [review]

File: alvr/server_core/src/web_server.rs (hunk 1 above)
Comment (The-personified-devil) [reply alvr/server_core/src/web_server.rs:270]: Especially since I don't quite trust the fs escape prevention code

**Pull request merged**